sound/soc/sof/ipc4-topology.c

```diff
@@ -1031,6 +1053,12 @@ static void sof_ipc4_unprepare_copier_module(struct snd_sof_widget *swidget)
 	pipeline->mem_usage = 0;
 
 	if (WIDGET_IS_AIF(swidget->id)) {
+		if (pipeline->use_chain_dma) {
+			pipeline->msg.primary &= ~SOF_IPC4_GLB_CHAIN_DMA_HOST_ID_MASK;
```

Comment:
why not pipeline->msg.primary = 0;  pipeline->msg.extension = 0; ? do we need to save any bits ?

Reply:
I was tempted to do that before, but did not because it leads to double zeroing of link dma id bits, when we earlier had only one pipeline for both the host and hda end. However, now that there is two pipelines and two msg -structs it makes sense to zero both in their appropriate places.